File: Cpp/TREE/Nodes without a Sibling.cpp
//https://www.geeksforgeeks.org/problems/print-all-nodes-that-dont-have-sibling/1
void sibl(Node *root,vector<int>&vec){
    if(root==NULL){
        return ;
    }
    if(root->left!=NULL && root->right==NULL){
        vec.push_back(root->left->data);
    }
     if(root->left==NULL && root->right!=NULL){
        vec.push_back(root->right->data);
    }
    sibl(root->left,vec);
    sibl(root->right,vec);
}
vector<int> noSibling(Node* root)
{
    // code here
    vector<int>vec;
    sibl(root,vec);
    if(vec.size()==0){
        vec.push_back(-1);
    }
    sort(vec.begin(),vec.end());
    return vec;
}

File: Python/TREE/Nodes without a Sibling.py
//https://www.geeksforgeeks.org/problems/print-all-nodes-that-dont-have-sibling/1
def help(root,arr):
    if root:
        if root.left and not root.right:
            arr.append(root.left.data)
        elif root.right and not root.left:
            arr.append(root.right.data)
        help(root.left,arr)
        help(root.right,arr)

def noSibling(root):
    arr=[]
    help(root,arr)
    arr.sort()
    return arr if arr else [-1]
